sdk/storage/azure-storage-file-datalake/azure/storage/filedatalake/_path_client.py

```diff
@@ -167,8 +167,10 @@ def _create_path_options(self, resource_type,
         if expires_on:
             try:
                 expires_on = convert_datetime_to_rfc1123(expires_on)
+                kwargs['expiry_options'] = 'Absolute'
             except AttributeError:
                 expires_on = str(expires_on)
+                kwargs['expiry_options'] = 'RelativeToNow'
```

Comment:
So the option `NeverExpire` never needs to be sent?

Reply:
From my understanding, the `BlobExpiryOptions` enums have `NeverExpire` mainly to be used for set APIs (which is also when exposing both `expiry_options` and `expires_on` to the cx makes sense to allow them that granularity when modifying the expiry options of an existing file). 

Since these are create APIs, the default behavior for creating a file is that it does not have an expiration time, and so only if the customer provides a value to `expires_on` then do we need to set `expiry_options` depending on what was provided for `expires_on`. Previously the critical code path for create APIs didn't let you set any expiry options, so the default behavior was no expiration.